src/Handlers.jl

```diff
@@ -110,37 +148,78 @@ function splitsegments(r::Router, h::Handler, segments)
     end
     return vals
 end
+function newsplitsegments(segments)
+    vals = Expr[]
+    for s in segments
+        if s == "*" #TODO: or variable, keep track of variable types and store in handler
+            T = Any
+        else
+            v = Val(Symbol(s))
+            T = typeof(v)
+        end
+        push!(vals, Expr(:(::), T))
+    end
+    return vals
+end
+
+function gethandler end
+gethandler(r::Router, args...) = r.default
 
-function register!(r::Router, method::DataType, scheme, host, path, handler)
+function register!(r::Router{id}, method, scheme, host, path, handler) where {id}
```

Comment:
How problematic would it be to change `register!` to a macro?
Do people run `register!` outside of Top-level scope?